io-engine/src/target/nvmf.rs

```diff
@@ -35,9 +35,14 @@ pub async fn unshare(uuid: &str) -> Result<(), NvmfError> {
 
 pub fn get_uri(uuid: &str) -> Option<String> {
     if let Some(ss) = NvmfSubsystem::nqn_lookup(uuid) {
-        // for now we only pop the first but we can share a bdev
-        // over multiple nqn's
-        ss.uri_endpoints().unwrap().pop()
+        // for now we only pop the most relevant but we can share a bdev
+        // over multiple nqn's. An nvmf+rdma uri will be prioritised for
+        // returning from here. If not present, then nvmf+tcp will be returned.
+        let mut uris = ss.uri_endpoints().expect("no uri endpoints");
+        let rdma_uri =
+            uris.iter().find(|u| u.starts_with("nvmf+RDMA")).cloned();
```

Comment:
Should this be lowercase?
```suggestion
            uris.iter().find(|u| u.starts_with("nvmf+rdma")).cloned();
```

Reply:
> Should this be lowercase?

The string constants in Mayastor and SPDK are capital case today so the `trstring` automatically comes in capital case. In some places, where required the case-insensitive comparison will be done.

Reply:
Could we pass `trstring` as lowercase ?
Otherwise we can modify the display in one of 2 ways:
1. generate the type string from the trtype using match
2. use tolower directly there

Reply:
I can do this in Formatter for `TransportId`. to_lowercase creates a new String, but should be ok.

```
        write!(
            f,
            "nvmf+{}://{}:{}",
            self.0.trstring.as_str().to_lowercase(),
            self.0.traddr.as_str(),
            self.0.trsvcid.as_str()
        )
    }
```